0000-interledger-protocol-4/0000-interledger-protocol-4.md

```diff
@@ -0,0 +1,159 @@
+- **Payment Channels, Not On-Ledger Escrow** - Since ILPv4 is optimized for smaller packets, speed and cost are of greater importance than in previous versions. ILPv4 uses ledgers or payment channels for settling bilateral payment obligations but ILPv4 packets are sent just between connectors, rather than through the underlying ledgers themselves. This enables packet timeouts to be short, because they do not need to include the processing time of slower ledgers, which further reduces [connector risks](../0018-connector-risk-mitigations/0018-connector-risk-mitigations.md).
+- **Forwarding, Not Delivery** - ILPv4 connectors forward packets based on their local exchange rates, in constrast with the first version of ILP in which connectors would attempt to deliver a fixed destination amount. Now, instead of fixed destination amount delivery being built into the core protocol, senders may use higher-level protocols to indicate the minimum amount the receiver should accept for a given packet and receivers can reject packets with less than that. This significantly simplifies the connector behavior, because the do not need to maintain up-to-date price information on the entire rest of the network and can simply apply their local rate instead. If receivers want to receive no more than a certain amount, they can reject packets that go too far over the amount and senders can retry the packet with lower amounts.
+- **Quoting is an Application Concern** - Connectors are only responsible for forwarding ILP packets and do not need to implement a [separate protocol for quoting](../0008-interledger-quoting-protocol/0008-interledger-quoting-protocol.md). Applications can use test packets to determine the exchange rate of a particular path.
+
+## Flow
+
+Before the protocol starts, the sender and receiver use a higher-level protocol to agree on the destination account and packet condition.
+
+1. Sender constructs an ILP Prepare packet with the receiver's destination account, the agreed-upon condition, and an amount and expiry of their choice. The sender may include additional data, the formatting of which is determined by the higher-level protocol.
+2. Sender sends the ILP Prepare packet to a connector over some communication channel, such as a Websocket. 
+3. Connector gets the packet and determines who the sender was based on the communication channel it was received through. The connector checks whether the sender has sufficient balance or credit to send the amount specified in the packet. If so, the connector reduces the sender's available balance by the value of the packet. If not, the connector returns an ILP Reject packet.
+4. Connector uses the destination ILP address from the packet and its local routing tables to determine which next hop to forward the packet to.
```

Comment:
Nitpick, but `uses the destination ILP address from the packet and its local routing tables` -> `uses its local routing tables and the destination ILP address from the packet` (otherwise you could technically read it as the packet's routing table)